### PR TITLE
feature/8051-add-spent-on

### PR DESCRIPTION
**Pull request by pandermatt**

### CODE REVIEW

**Comment by pandermatt** [comment]
I discussed the solution with @Kjir (and the tests are green)

File: spec/assigner_spec.rb [hunk 1]
```diff
@@ -4,15 +4,17 @@
   context 'named special cases' do
     context 'weekly meetings' do
       it 'should assign the meetings' do
-        a = Assigner.new '92345678: internal weekly meeting'
+        a = Assigner.new '92345678: internal weekly meeting : : 2017-04-25T09:02:30+00: 00'
```
Comment (lukaselmer) [inline spec/assigner_spec.rb:7]: @pascalandermatt if you select a tag int toggle, `: :` will not be in the string. e.g.

`571115323: bla bla bla : Development : 2017-04-26T08:05:00+00: 00`

Comment (lukaselmer) [reply spec/assigner_spec.rb:7]: image

=> just use the last part as date

**Review by lukaselmer — CHANGES_REQUESTED** [review]
please also make it work if one selects a tag in toggl

File: spec/redmine_adapter_mock.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class RedmineAdapterMock
   #<TimeEntry id: 1815, project_id: 24, user_id: 1, issue_id: 916, hours: 0.2325, comments: "84561435: Intern - Sitzung - Nächste Sitzung vorber...", activity_id: 45, spent_on: "2013-07-30", tyear: 2013, tmonth: 7, tweek: 31, created_on: "2013-07-30 14:40:32", updated_on: "2013-07-30 14:40:32">
-  TimeEntryMock = Struct.new(:issue_id, :comments, :activity_id, :hours, :id, :saved) do
+  TimeEntryMock = Struct.new(:issue_id, :spent_on, :comments, :activity_id, :hours, :id, :saved) do
```
Comment (lukaselmer) [inline spec/redmine_adapter_mock.rb:3]: why do you add `spent_on` in front? if you add it at the end, you do not need to change any of the mocks below, right?

File: spec/assigner_spec.rb [hunk 3]
```diff
@@ -4,15 +4,17 @@
...
         a.issue_ids.first.should == 1339
         a.issue_ids.count.should == 1
+        a.spent_on.should == Date.new(2017, 4, 25)
```
Comment (lukaselmer) [inline spec/assigner_spec.rb:10]: please test the spent_on feature separately

File: spec/assigner_spec.rb [hunk 4]
```diff
@@ -29,7 +31,7 @@
   end
 
   it 'should assign ticket 1' do
-    a = Assigner.new '92345678: 741 Bla bla bla'
+    a = Assigner.new '92345678: 741 Bla bla bla : : 2017-04-25T09:02:30+00: 00'
```
Comment (lukaselmer) [inline spec/assigner_spec.rb:34]: all these do not need to change, right?

File: spec/assigner_spec.rb [hunk 5]
```diff
@@ -128,8 +130,21 @@
...
+      a = Assigner.new '92345678: Redmine Auto Time Entries : Entwicklung : : 2017-04-25T09:02:30+00: 00'
+      a.spent_on.should == Date.new(2017, 4, 25)
+    end
+
+    it 'does not hurt old data' do
+      a = Assigner.new '92345678: Redmine Auto Time Entries : Entwicklung'
+      a.spent_on.should == Date.today
+    end
+  end
+
```
Comment (lukaselmer) [inline spec/assigner_spec.rb:149]: blank line

File: lib/assigner.rb [hunk 6]
```diff
@@ -19,6 +20,13 @@ def check_special_case(mapper)
     mapper.map!(self)
   end
 
+  def extract_end_hours
+    self.spent_on = Date.today
```
Comment (lukaselmer) [inline lib/assigner.rb:24]: this is done automatically, so we do not need this

**Comment by coorasse** [comment]
I'm happy to leave this to you and Lukas. I do not need to place a review. Thanks both of you!

**Comment by lukaselmer** [comment]
See #3

**Pull request closed without merging**